engine/gamesys/src/gamesys/scripts/script_image.cpp

```diff
@@ -294,33 +294,17 @@ namespace dmGameSystem
 
             PushImageParameters(L, image);
 
-            uint32_t datasize = bytes_per_pixel * image.m_Width * image.m_Height;
+            uint32_t imagesize = image.m_Width * image.m_Height;
+            uint32_t datasize = bytes_per_pixel * imagesize;
 
             lua_pushliteral(L, "buffer");
 
-            uint8_t num_components = 1;
-            switch(image.m_Type)
-            {
-                case dmImage::TYPE_RGB:
-                    num_components = 3;
-                    break;
-                case dmImage::TYPE_RGBA:
-                    num_components = 4;
-                    break;
-                case dmImage::TYPE_LUMINANCE:
-                    num_components = 1;
-                    break;
-                case dmImage::TYPE_LUMINANCE_ALPHA:
-                    num_components = 2;
-                default:break;
-            }
-
             dmBuffer::StreamDeclaration streams_decl[] = {
-                { dmHashString64("data"), dmBuffer::VALUE_TYPE_UINT8, num_components }
+                { dmHashString64("data"), dmBuffer::VALUE_TYPE_UINT8, bytes_per_pixel }
             };
 
             dmBuffer::HBuffer buffer = 0;
-            dmBuffer::Create(datasize, streams_decl, 1, &buffer);
+            dmBuffer::Create(imagesize, streams_decl, 1, &buffer);
```

Comment:
The issue here. Buffer size should be `imagesize` not `datasize`, because `bytes_per_pixel` already taken into account in stream.